src/include/planner/plannodes/abstract_scan_plan_node.h

```diff
@@ -57,6 +57,24 @@ class AbstractScanPlanNode : public AbstractPlanNode {
       return *dynamic_cast<ConcreteType *>(this);
     }
 
+    /**
+     * @param limit number of tuples to limit to
+     * @return builder object
+     */
+    ConcreteType &SetScanLimit(uint32_t limit) {
+      scan_limit_ = limit;
+      return *dynamic_cast<ConcreteType *>(this);
+    }
+
+    /**
+     * @param offset offset for where to limit from
```

Comment:
This is related to the `LIMIT` clause in SQL:

https://www.postgresql.org/docs/current/queries-limit.html

Modify the function header to be like this:

```
  /**
   * The number of tuples that this scan should emit due to a LIMIT clause.
   * You must check the 'has limit' flag first to determine whether the value
   * in this field should actually be used. Otherwise it should be ignored.
   * @return number to limit to
   */
```

Reply:
This is done. I hope you don't mind but I used the `@pre` tag, I thought it would be a nice touch. Happy to remove it though.